src/node/connection.rs

```diff
@@ -31,8 +36,12 @@ impl NodeClient {
             .await
             .inspect_err(|e| error!("Failed to acquire a statequery client: {:?}", e))?;
 
-        // Run the action and ensure the client is released afterwards
-        let result = action(client).await;
+        // Run the action with a timeout
+        let result = timeout(duration, action(client)).await.map_err(|_| {
+            let msg = format!("Timeout after {} seconds", duration.as_secs());
+            error!("{} in with_statequery_timeout", msg);
+            BlockfrostError::timeout(msg)
+        })?;
```

Comment:
This could be just slightly clearer. We have a `Result<Result<A, Err>, tokio::Elapsed>`, and we apply the `?` only on the top level result, because then such a timed out connection is useless in the future, as it’s probably in an incoherent state.

Hmmm… maybe a comment? 👀